usecases/schema/schema_comparison.go

```diff
@@ -0,0 +1,268 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package schema
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/usecases/sharding"
+)
+
+// Diff creates human-readable information about the difference in two schemas,
+// returns a len=0 slice if schemas are identical
+func Diff(
+	leftLabel string, left *State,
+	rightLabel string, right *State,
+) []string {
+	var msgs []string
+
+	if len(left.ObjectSchema.Classes) != len(right.ObjectSchema.Classes) {
+		msg := fmt.Sprintf("%s has %d classes, but %s has %d classes",
+			leftLabel, len(left.ObjectSchema.Classes),
+			rightLabel, len(right.ObjectSchema.Classes))
+		msgs = append(msgs, msg)
+	}
+
+	leftClasses := map[string]*models.Class{}
+	rightClasses := map[string]*models.Class{}
+
+	for _, class := range left.ObjectSchema.Classes {
+		leftClasses[class.Class] = class
+	}
```

Comment:
can be merged with `for className, classLeft := range leftClasses {` loop as in `propsComparison` to finally have 3 loops instead of 4